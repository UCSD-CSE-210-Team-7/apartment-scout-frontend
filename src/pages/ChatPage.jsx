import React from "react";
import Sidebar from "../components/ChatInterface/Sidebar";
import Chat from "../components/ChatInterface/Chat";
import "../styles/chat-styles.scss";

const ChatPage = () => {
  return (
    <div className="home">
      <div className="container">
        <Sidebar />
        <Chat />
      </div>
    </div>
  );
};

export default ChatPage;
