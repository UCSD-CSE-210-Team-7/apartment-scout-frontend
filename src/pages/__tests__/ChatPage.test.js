import { render as _r, screen, act, fireEvent, waitForElementToBeRemoved, waitFor } from '@testing-library/react'
import { render } from '../../test_setup'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'
import { MemoryRouter, BrowserRouter, RouterProvider, createMemoryRouter, createRoutesFromElements, Routes, Route } from 'react-router-dom';

import { data } from '../../dummy_data/me.json'
import ChatPage, { QUERY_CONVERSATIONS, SEND_MESSAGE, MESSAGE_SUBSCRIPTION }  from '../ChatPage';

beforeEach( () => {
  window.HTMLElement.prototype.scrollIntoView = function() {};
  dateNowSpy = jest.spyOn(Date.prototype, 'toLocaleString').mockImplementation(() => 'dummy date');
})

test("render with no user", () => {
  const { asFragment } = render( 
    <ChatPage/>,
    { mocks: [ 
      { 
        request: { 
          query: QUERY_CONVERSATIONS, 
          variables: {conversation_id: undefined}
        },
        result: { data }
      },
      { 
        request: { query: MESSAGE_SUBSCRIPTION }, 
        result: { message: data.me.conversations[0].messages[0] }
      }
    ] }
  );
  expect(asFragment()).toMatchSnapshot()
});

test("render after loading", async () => {
  const { asFragment } = render( 
    <ChatPage/>,
    { mocks: [ 
      { 
        request: { 
          query: QUERY_CONVERSATIONS, 
          variables: {conversation_id: undefined}
        },
        result: { data }
      },
      { 
        request: { query: MESSAGE_SUBSCRIPTION }, 
        result: { data: { message: null }}
      }
    ] }
  );
  await waitForElementToBeRemoved(screen.getByText('Loading...'));
  expect(asFragment()).toMatchSnapshot()
});


test("click a user", async () => {
  const { asFragment } = render( 
    <ChatPage/>,
    { mocks: [ 
      { 
        request: { 
          query: QUERY_CONVERSATIONS, 
          variables: {conversation_id: undefined}
        },
        result: { data }
      },
      { 
        request: { query: MESSAGE_SUBSCRIPTION }, 
        result: { data: { message: null}}
      }
    ] }
  );
  await waitForElementToBeRemoved(screen.getByText('Loading...'));
  act(() => {
    userEvent.click(screen.getByText('Manasi Agrawal'));
  })
  expect(asFragment()).toMatchSnapshot()
});


test("click user, subscribed", async () => {
  const { asFragment } = render( 
    <ChatPage/>,
    { mocks: [ 
      { 
        request: { 
          query: QUERY_CONVERSATIONS, 
          variables: {conversation_id: undefined}
        },
        result: { data }
      },
      { 
        request: { query: MESSAGE_SUBSCRIPTION }, 
        result: { data: { message: {
            "msg_id": -1,
            "msg_text": "Hi. This is Shubham.",
            "msg_time": "2023-03-10T15:35:13.289Z",
            "sender": {
              "name": "Shubham Kulkarni",
              "email": "skulkarn@ucsd.edu"
            },
            "conversation": {
              "conversation_id": 1,
              "person_a": {
                "email": "skulkarn@ucsd.edu",
                "name": "Shubham Kulkarni"
              },
              "person_b": {
                "email": "maagrawa@ucsd.edu",
                "name": "Manasi Agrawal"
              }
            }
        }}}
      }
    ] }
  );
  await waitForElementToBeRemoved(screen.getByText('Loading...'));
  act(() => {
    userEvent.click(screen.getByText('Manasi Agrawal'));
  })
  expect(asFragment()).toMatchSnapshot()
});
/*
test("render with user and new message", async () => {
  const { asFragment } = render( 
    <Chat conversation={data.messages[0].conversation} user={data.messages[0].sender} />, 
    { mocks: [ 
      { 
        request: { 
          query: QUERY_MESSAGES, 
          variables: {conversation_id: data.messages[0].conversation.conversation_id}}, 
        result: { data }
      },
      { 
        request: { query: MESSAGE_SUBSCRIPTION }, 
        result: { data: { message: {
          "msg_id": 9,
          "msg_text": "Additional Message",
          "msg_time": "2023-02-28T13:58:49.334Z",
          "sender": {
            "name": "Shubham Kulkarni",
            "email": "skulkarn@ucsd.edu"
          },
          "conversation": {
            "conversation_id": 1,
            "person_a": {
              "email": "skulkarn@ucsd.edu",
              "name": "Shubham Kulkarni"
            },
            "person_b": {
              "name": "Test User",
              "email": "test@gmail.com"
            }
          }
        }
        } }
      }
    ] }
  );
  await waitForElementToBeRemoved(screen.getByText('Loading...'));
  expect(asFragment()).toMatchSnapshot()
});

test("send new message", async () => {
  const msg_text = "test message"
  const { asFragment } = render( 
    <Chat conversation={data.messages[0].conversation} user={data.messages[0].sender} />, 
    { mocks: [ 
      { 
        request: { 
          query: QUERY_MESSAGES, 
          variables: {conversation_id: data.messages[0].conversation.conversation_id}}, 
        result: { data }
      },
      { 
        request: { query: MESSAGE_SUBSCRIPTION }, 
        result: { data: { message: null } } 
      },
      { 
        request: { 
          query: SEND_MESSAGE, 
          variables: { msg_text, conversation: data.messages[0].conversation.conversation_id}
        }, 
        result: { 
          data: { 
            createMessage: { 
              ...data.messages[2],
            } 
          } 
        } 
      },
      { 
        request: { 
          query: QUERY_MESSAGES, 
          variables: {conversation_id: data.messages[0].conversation.conversation_id}}, 
        result: { data: { messages: [
          ...data.messages, 
          {
            ...data.messages[0],
            msg_id: -1,
            msg_text,
          }
        ] } }
      },
    ] }
  );
  await waitForElementToBeRemoved(screen.getByText('Loading...'));
  const textbox = screen.getByRole('textbox')
  fireEvent.change(textbox, {target: { value: msg_text }})
  fireEvent.click(screen.getByAltText('send-button'))
  await waitFor( () => screen.findByText(msg_text));
  expect(asFragment()).toMatchSnapshot()
});
*/
