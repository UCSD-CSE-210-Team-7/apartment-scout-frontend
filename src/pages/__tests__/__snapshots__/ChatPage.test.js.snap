// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`click a user 1`] = `
<DocumentFragment>
  <div
    style="display: flex; flex-grow: 1; max-height: 90vh;"
  >
    <div
      style="background: rgb(217, 217, 217); width: 20em;"
    >
      <div
        class="userChat"
        style="background: rgb(173, 232, 244);"
      >
        <img
          alt=""
          src="user.png"
        />
        <div
          class="userChatInfo"
        >
          <span>
            Manasi Agrawal
          </span>
          <p>
            dsafdsafa
          </p>
        </div>
      </div>
      <div
        class="userChat"
      >
        <img
          alt=""
          src="user.png"
        />
        <div
          class="userChatInfo"
        >
          <span>
            Shubham Kulkarni
          </span>
          <p>
            dsafdsaf
          </p>
        </div>
      </div>
      <div
        class="userChat"
      >
        <img
          alt=""
          src="user.png"
        />
        <div
          class="userChatInfo"
        >
          <span>
            Guy Nir
          </span>
          <p>
            bnvcncvb
          </p>
        </div>
      </div>
      <div
        class="userChat"
      >
        <img
          alt=""
          src="user.png"
        />
        <div
          class="userChatInfo"
        >
          <span>
            Ajinkya Bokade
          </span>
          <p>
            sfd
          </p>
        </div>
      </div>
    </div>
    <div
      style="display: flex; flex-direction: column; background: rgb(245, 245, 245); font-size: 1.5em; flex-grow: 1; justify-content: center;"
    >
      <div
        style="background: white; display: flex; align-items: center; justify-content: center; padding: 1em; color: black;"
      >
        <img
          alt="profile-pic"
          class="contact-photo"
          src="user.png"
        />
        <span>
          Shubham Kulkarni
        </span>
      </div>
      <div
        style="display: flex; flex-direction: column; flex-grow: 2; overflow-y: scroll;"
      >
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            Hi. This is Shubham.
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            Hi Shubham. This is Manasi.
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            Hi Manasi. How can I help you today?
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            I need a tour for 625 8th Avenue New York NY. Please get this done ASAP.
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            On my way now.
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            Tour is complete. Please leave a review!
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            Thanks for your work!
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            Thanks Manasi!
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            great
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            wonderful
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            excellent
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            hi
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            sd
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            sdfds
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            s
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            sdfsd
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            dsafsda
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            asdfsa
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            sdafsdaf
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            dsafdsafa
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div />
      </div>
      <div
        style="display: flex; padding: 0.5em;"
      >
        <input
          placeholder="Type..."
          style="flex-grow: 1; border: 0.5px solid black; background: rgb(245, 245, 245); border-radius: 2em; padding: 0px 1em;"
          type="text"
          value=""
        />
        <img
          alt="send-button"
          src="send_icon.png"
          style="height: 2em; width: 2em; margin: 0em 1em;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`click user, subscribed 1`] = `
<DocumentFragment>
  <div
    style="display: flex; flex-grow: 1; max-height: 90vh;"
  >
    <div
      style="background: rgb(217, 217, 217); width: 20em;"
    >
      <div
        class="userChat"
      >
        <img
          alt=""
          src="user.png"
        />
        <div
          class="userChatInfo"
        >
          <span>
            Shubham Kulkarni
          </span>
          <p>
            dsafdsaf
          </p>
        </div>
      </div>
      <div
        class="userChat"
      >
        <img
          alt=""
          src="user.png"
        />
        <div
          class="userChatInfo"
        >
          <span>
            Guy Nir
          </span>
          <p>
            bnvcncvb
          </p>
        </div>
      </div>
      <div
        class="userChat"
      >
        <img
          alt=""
          src="user.png"
        />
        <div
          class="userChatInfo"
        >
          <span>
            Ajinkya Bokade
          </span>
          <p>
            sfd
          </p>
        </div>
      </div>
      <div
        class="userChat"
        style="background: rgb(173, 232, 244);"
      >
        <img
          alt=""
          src="user.png"
        />
        <div
          class="userChatInfo"
        >
          <span>
            Manasi Agrawal
          </span>
          <p>
            dsafdsafa
          </p>
        </div>
      </div>
    </div>
    <div
      style="display: flex; flex-direction: column; background: rgb(245, 245, 245); font-size: 1.5em; flex-grow: 1; justify-content: center;"
    >
      <div
        style="background: white; display: flex; align-items: center; justify-content: center; padding: 1em; color: black;"
      >
        <img
          alt="profile-pic"
          class="contact-photo"
          src="user.png"
        />
        <span>
          Shubham Kulkarni
        </span>
      </div>
      <div
        style="display: flex; flex-direction: column; flex-grow: 2; overflow-y: scroll;"
      >
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            Hi. This is Shubham.
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            Hi Shubham. This is Manasi.
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            Hi Manasi. How can I help you today?
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            I need a tour for 625 8th Avenue New York NY. Please get this done ASAP.
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            On my way now.
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            Tour is complete. Please leave a review!
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            Thanks for your work!
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            Thanks Manasi!
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            great
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            wonderful
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            excellent
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            hi
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            sd
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            sdfds
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            s
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            sdfsd
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            dsafsda
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            asdfsa
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            sdafsdaf
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            dsafdsafa
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div
          style="list-style-type: none; max-width: 80%; margin: 0.5em 2em; border-radius: 1em; padding: 1em; box-shadow: 0 4px 4px #00000040; display: flex; flex-direction: column; background: white;"
        >
          <p
            style="margin: 0px;"
          >
            Hi. This is Shubham.
          </p>
          <span
            style="font-size: 0.5em; align-self: flex-end;"
          >
            dummy date
          </span>
        </div>
        <div />
      </div>
      <div
        style="display: flex; padding: 0.5em;"
      >
        <input
          placeholder="Type..."
          style="flex-grow: 1; border: 0.5px solid black; background: rgb(245, 245, 245); border-radius: 2em; padding: 0px 1em;"
          type="text"
          value=""
        />
        <img
          alt="send-button"
          src="send_icon.png"
          style="height: 2em; width: 2em; margin: 0em 1em;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`render after loading 1`] = `
<DocumentFragment>
  <div
    style="display: flex; flex-grow: 1; max-height: 90vh;"
  >
    <div
      style="background: rgb(217, 217, 217); width: 20em;"
    >
      <div
        class="userChat"
      >
        <img
          alt=""
          src="user.png"
        />
        <div
          class="userChatInfo"
        >
          <span>
            Manasi Agrawal
          </span>
          <p>
            dsafdsafa
          </p>
        </div>
      </div>
      <div
        class="userChat"
      >
        <img
          alt=""
          src="user.png"
        />
        <div
          class="userChatInfo"
        >
          <span>
            Shubham Kulkarni
          </span>
          <p>
            dsafdsaf
          </p>
        </div>
      </div>
      <div
        class="userChat"
      >
        <img
          alt=""
          src="user.png"
        />
        <div
          class="userChatInfo"
        >
          <span>
            Guy Nir
          </span>
          <p>
            bnvcncvb
          </p>
        </div>
      </div>
      <div
        class="userChat"
      >
        <img
          alt=""
          src="user.png"
        />
        <div
          class="userChatInfo"
        >
          <span>
            Ajinkya Bokade
          </span>
          <p>
            sfd
          </p>
        </div>
      </div>
    </div>
    <div
      style="display: flex; flex-direction: column; background: rgb(245, 245, 245); font-size: 1.5em; flex-grow: 1; justify-content: center;"
    >
      <h1
        style="align-self: center;"
      >
        Select a conversation
      </h1>
    </div>
  </div>
</DocumentFragment>
`;

exports[`render with no user 1`] = `
<DocumentFragment>
  <div
    style="display: flex; flex-grow: 1; max-height: 90vh;"
  >
    <h1>
      Loading...
    </h1>
    <div
      style="display: flex; flex-direction: column; background: rgb(245, 245, 245); font-size: 1.5em; flex-grow: 1; justify-content: center;"
    >
      <h1
        style="align-self: center;"
      >
        Select a conversation
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
